### PR TITLE
Fixes a bug where there is a missing await.

### PR DESCRIPTION
**Pull request by Falven**
## Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- [X] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [X] I have performed a self-review of my own code

### CODE REVIEW

File: llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py [hunk 1]
```diff
@@ -212,9 +213,10 @@ async def aput(
         table_name = (
             DEFAULT_COLLECTION if not collection else sanitize_table_name(collection)
         )
-        await self._atable_service_client.create_table_if_not_exists(
-            table_name
-        ).upsert_entity(
+        atable_service_client = (
```
Comment (nerdai) [inline llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py:216]: we should call this `atable_client`

File: llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py [hunk 2]
```diff
@@ -185,7 +185,8 @@ def put(
         table_name = (
             DEFAULT_COLLECTION if not collection else sanitize_table_name(collection)
         )
-        self._table_client.create_table_if_not_exists(table_name).upsert_entity(
+        table_client = self._table_client.create_table_if_not_exists(table_name)
```
Comment (nerdai) [inline llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py:188]: unfortunately, it looks like `self._table_client` should be `self._table_service_client`

**Review by nerdai — COMMENTED** [review]
thanks @Falven -- left a couple of comments

File: llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py (hunk 1 above)
Comment (Falven) [reply llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py:216]: Thanks for reviewing @nerdai 
I disagree. Those are two different concepts/objects. Table client vs table service client.

Comment (nerdai) [reply llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py:216]: Right. Unless I read this resource incorrectly:

https://learn.microsoft.com/en-us/python/api/overview/azure/data-tables-readme?view=azure-python#creating-a-table

The table_service_client allows us to create a table and returns a table_client

Comment (Falven) [reply llama-index-integrations/storage/kvstore/llama-index-storage-kvstore-azure/llama_index/storage/kvstore/azure/base.py]: Normalized the naming.

**Review by nerdai — APPROVED** [review]
LGTM @Falven -- thanks again!

**Pull request merged**